ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -469,11 +475,22 @@ private void onResponse(List<ResponseInfo> responseInfoList) {
      */
     @Override
     public void run() {
+      long startTime;
+      long lastTime;
       try {
         while (isOpen.get()) {
+          startTime = time.milliseconds();
+          lastTime = startTime;
           List<RequestInfo> requestInfoList = pollForRequests();
+          routerMetrics.operationControllerPollforRequestTimeMs.update(time.milliseconds() - lastTime);
+          lastTime = time.milliseconds();
           List<ResponseInfo> responseInfoList = networkClient.sendAndPoll(requestInfoList);
+          routerMetrics.numOfRequestInfosPerPoll.update(requestInfoList.size());
+          routerMetrics.operationControllerSendAndPollTimeMs.update(time.milliseconds() - lastTime);
```

Comment:
For my next commit I will follow your suggestion and drop this metric to make metrics number small. We can discuss more offline.